src/reducers/GraphDataState.ts

```diff
@@ -39,127 +40,171 @@ export const INITIAL_GRAPH_STATE: GraphState = {
+        filterState: updateState(state.filterState, {
+          showVirtualServices: !state.filterState.showVirtualServices
+        })
+      });
     case getType(GraphFilterActions.toggleGraphMissingSidecars):
-      newState.filterState.showMissingSidecars = !state.filterState.showMissingSidecars;
-      break;
+      return updateState(state, {
+        filterState: updateState(state.filterState, {
+          showMissingSidecars: !state.filterState.showMissingSidecars
+        })
+      });
     case getType(GraphFilterActions.toggleGraphSecurity):
-      newState.filterState.showSecurity = !state.filterState.showSecurity;
-      break;
+      return updateState(state, {
+        filterState: updateState(state.filterState, {
+          showSecurity: !state.filterState.showSecurity
+        })
+      });
     case getType(GraphFilterActions.toggleLegend):
-      newState.filterState.showLegend = !state.filterState.showLegend;
-      break;
+      return updateState(state, {
+        filterState: updateState(state.filterState, {
+          showLegend: !state.filterState.showLegend
+        })
+      });
     case getType(GraphFilterActions.toggleServiceNodes):
-      newState.filterState.showServiceNodes = !state.filterState.showServiceNodes;
-      // TODO: This should be handled in GraphPage.ComponentDidUpdate (Init graph on serviceNodeschange)
-      newState.graphData = INITIAL_GRAPH_STATE.graphData;
-      newState.graphDataDuration = INITIAL_GRAPH_STATE.graphDataDuration;
-      newState.graphDataTimestamp = INITIAL_GRAPH_STATE.graphDataTimestamp;
-      newState.summaryData = INITIAL_GRAPH_STATE.summaryData;
-      break;
+      return updateState(state, {
+        filterState: updateState(state.filterState, {
+          showServiceNodes: !state.filterState.showServiceNodes
+        }),
+        // TODO: This should be handled in GraphPage.ComponentDidUpdate (Init graph on type change)
+        graphData: INITIAL_GRAPH_STATE.graphData,
+        graphDataDuration: INITIAL_GRAPH_STATE.graphDataDuration,
+        graphDataTimestamp: INITIAL_GRAPH_STATE.graphDataTimestamp,
+        summaryData: INITIAL_GRAPH_STATE.summaryData
+      });
     case getType(GraphFilterActions.toggleTrafficAnimation):
-      newState.filterState.showTrafficAnimation = !state.filterState.showTrafficAnimation;
-      break;
+      return updateState(state, {
+        filterState: updateState(state.filterState, {
+          showTrafficAnimation: !state.filterState.showTrafficAnimation
+        })
+      });
     case getType(GraphFilterActions.toggleUnusedNodes):
-      newState.filterState.showUnusedNodes = !state.filterState.showUnusedNodes;
-      break;
+      return updateState(state, {
+        filterState: updateState(state.filterState, {
+          showUnusedNodes: !state.filterState.showUnusedNodes
+        })
+      });
     default:
-      break;
+      // Return unmodified state if there are no changes.
+      return state;
   }
-
-  return newState;
```

Comment:
to clarify, this is needed, but regardless this change, we should include it.
This reducer was returning a new state even when there wasn't any change or what is worst, even on unrelated actions.
This was confusing / making fail the "persist: hydrate" action.